### PR TITLE
[MSM] fix dt_model in bayesian msm, added tests for other estimators

### LINKED ISSUE #1114: [MSM.dt_model] : not always set automatically -> dangerous consequences
Please read in full (and preferably on the web). 

Reproducing this takes quite some time and I want to hear your thoughts before pulling newest devel and trying to reproduce.

It seems that the attribute 'dt_model' is not correctly set to the MSM.lag value in "some" of the objects in its.models when using the Bayesian estimator. This leads to dangerous behavior that @fabian-paul and me had observed (and fixed) already (https://github.com/markovmodel/PyEMMA/pull/929 and https://github.com/markovmodel/PyEMMA/issues/789). 

It is **not** visible at the level of the its object and stuff derived from its its.timescales attribute, most notably, the ITS-plot. This explains why it's hard to detect. 

It only surfaces when dealing with the models directly, i.e. the its.model-objects and the stuff derived from them, like mfpts and the object's own msm.timescales

We start with:
```python
print(ML_its)
ImpliedTimescales(estimator=MaximumLikelihoodMSM(connectivity='largest', count_mode='sliding',
           dt_traj='1 step', lag=1, maxerr=1e-08, maxiter=1000000,
           reversible=True, score_k=10, score_method='VAMP2', sparse=False,
           statdist_constraint=None),
         lags=array([ 20,  22,  24,  26,  28,  30,  40,  50,  60,  70,  80,  90, 100,
       110, 120, 140, 160, 200, 240, 280]),
         n_jobs=1, nits=7, show_progress=True)

print(B_its)
ImpliedTimescales(estimator=BayesianMSM(conf=0.95, connectivity='largest', count_mode='effective',
      dt_traj='1 step', lag=1, nsamples=50, nsteps=None, reversible=True,
      show_progress=True, sparse=False, statdist_constraint=None),
         lags=array([ 20,  22,  24,  26,  28,  30,  40,  50,  60,  70,  80,  90, 100,
       110, 120, 140, 160, 200, 240, 280]),
n_jobs=1, nits=7, show_progress=True)
```

Nothing can be seen at the level of its.timscales:

```python 
print("ITS from the its object with ML-MSM estimator:")
print(ML_its.timescales[:n,0].round())
[  619150.   714074.   765038.   753040.   863927.   843592.  1111366.
  1330246.  1554922.  1722230.  2011382.]
print("ITS from the its object with Bayes-MSM estimator:")
print(B_its.timescales[:n,0].round())
[  619411.   714348.   765308.   753293.   864199.   843845.  1111669.
  1330609.  1555347.  1721472.  2010600.
```
or in the plots:

snap23

However, looking at the dt_model atribute of the models underneath it...
```python
for ii in range(n):
    ML_MSM = ML_its.models[ii]
    B_MSM = B_its.models[ii]
    print(ML_MSM.dt_model, B_MSM.dt_model)
20.0 step 20.0 step
22.0 step 22.0 step
24.0 step 24.0 step
26.0 step 26.0 step
28.0 step 28.0 step
30.0 step 30.0 step
40.0 step 40.0 step
50.0 step 50.0 step
60.0 step 60.0 step
70.0 step 70.0 step
80.0 step 1 step
```
Look at the last line. 
(i cropped the list, it's not associated with being the last element). 

I'm guessing the problem comes from the dt_model attribute because the MSM.lag is correctly set in both:
```python
for ii in range(n):
    ML_MSM = ML_its.models[ii]
    B_MSM = B_its.models[ii]
    print(ML_MSM.lag, B_MSM.lag)
20 20
22 22
24 24
26 26
28 28
30 30
40 40
50 50
60 60
70 70
80 80
```
What set off the alarm was:
```python
n = 11
ML_MSM = ML_its.models[ii]
B_MSM = B_its.models[ii]
# Slowest timescales in both cases
print(ML_MSM.timescales()[0])
# 2011381.67352
print(B_MSM.timescales()[0])
# 25132.497453 (ca. 2 orders of mag (80 steps, see above) smaller)

# MFPTs 1->7 for both MLE and Bayes
print(pyemma.msm.tpt(ML_MSM, [1], 
                     [7]).mfpt)
# 5951972.75995
print(pyemma.msm.tpt(B_MSM, [1], 
                     [7]).mfpt)
# 74442.8420277 # Same thing...ca. 80 times faster
```
The "wrong" values multiplied by the ML_MSM.lagtime value will recover the right B_MSM values. 

I hope the lengthy description is helpful, because I cannot get any further into the estimator/model structure.

### PR DESCRIPTION
**Pull request by marscher**
Fixes #1114, #1083

### CODE REVIEW

File: pyemma/msm/estimators/maximum_likelihood_msm.py [hunk 1]
```diff
@@ -148,6 +148,7 @@ def __init__(self, lag=1, reversible=True, count_mode='sliding', sparse=False,
         # time step
         self.dt_traj = dt_traj
         self.timestep_traj = _TimeUnit(dt_traj)
+        self.dt_model = self.dt_traj
```
Comment (gph82) [inline pyemma/msm/estimators/maximum_likelihood_msm.py:152]: hI @marscher  I can't really control much, but at what other instance was dt_model being set in the models that HAD it set?

**Comment by marscher** [comment]
@gph82 it would be nice if you could have a look.

**Comment by gph82** [comment]
I thought I had started a review thingy, did you see that?

**Comment by marscher** [comment]
no, there is nothing there.

**Comment by gph82** [comment]
ok now it's there, I had one last click to make

**Comment by marscher** [comment]
going to merge this now, because it definitely fixes a bug with dt_traj handling in BMSM. However I can not prove that it solves the issue you've described @gph82, because I could not reproduce it.

**Pull request merged**